cmd/otel-allocator/README.md

```diff
@@ -0,0 +1,82 @@
+# Target Allocator
+
+The TargetAllocator is an optional separately deployed component of an OpenTelemetry Collector setup, which is used to
+distribute targets of the PrometheusReceiver on all deployed Collector instances.
+
+# Design
+
+If the Allocator is activated, all Prometheus configurations will be transferred in a separate ConfigMap which get in
+turn mounted to the Allocator.    
+This configuration will be resolved to target configurations and then split across all OpenTelemetryCollector instances.
+
+TargetAllocators exposes the results as [HTTP_SD endpoints](https://prometheus.io/docs/prometheus/latest/http_sd/)
```

Comment:
How does the OTEL colletor get the final config (resolved targets)? 

Does the OTEL operator configure the collector to use the http_sd to get the targets?

Reply:
Yes, the OTEL operator rewrites the collector `prometheus` receiver config to use `http_sd` for all configured jobs, pointing at the target allocator instance for that collector.

There is still a gap, that @secustor has proposed to close with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8055.  This addresses the fact that the Prometheus operator CRs can cause new jobs to be exposed at the target allocator but the collector configuration is static and does not currently have a mechanism for discovering those jobs.  The target allocator exposes a job list resource that should suffice to discover new/ended jobs.

Reply:
Yes, that's correct. The operator configures the collector to use http_sd endpoints which are provided by the TargetAllocator.

Currently the Operator defines the p8s jobs which the collector should query, therefore no p8s operator CR defined jobs are added to the collector. Later on the the collector should query the jobs directly from the TargetAllocator. 
See following issues: 
- operator: https://github.com/open-telemetry/opentelemetry-operator/issues/713
- collector: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7944